tests/functional/regressions/test_issue167.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import unicode_literals
+
+from textx import metamodel_from_str, get_model, get_metamodel
+from pytest import raises
+from textx.exceptions import TextXSemanticError
+
+"""
+This test and example demonstrates how to use a custom scope provider to
+create an object if a referenced object is not found in the model. See #167.
+
+Here we can
+ * define persons and
+ * we can specify that persons know each other
+
+The goal is that we can use a custom scope provider to define a known
+person on the fly, if it is not define.
+
+Note: This example has some limits. Another scope provider which requires to
+reference a person not yet created, may get an error.
+"""
+
+grammar = r'''
+Model: (persons+=Person|knows+=Knows)*;
+Person: ':' 'person' name=ID;
+Knows: person1=[Person] 'knows' person2=[Person];
+'''
+
+
+def person_definer_scope(knows, attr, attr_ref):
+    m = get_model(knows)  # get the model of the currently processed element
+    name = attr_ref.obj_name  # the name of currently looked up element
+    found_persons = list(filter(lambda p: p.name == name, m.persons))
+    if len(found_persons) > 0:
+        return found_persons[0]  # if a person exists, return it
+    else:
+        mm = get_metamodel(m)  # else, create it and store it in the model
+        person = mm['Person']()
```

Comment:
Is this ok?
Is it ok to instantiate a model object that way? We could maybe add something in the docs for that...

Reply:
I think this style of instantiation is perfectly fine.